docs_src/design/ucr/Extending-Device-Data.md

```diff
@@ -0,0 +1,78 @@
+and the original south-bound Device Resources; that is, to indicate that these Resources "extend" the original Device data.
+
+It is desirable that the means of conveying this information become standardized for those EdgeX microservices
+which provide and use it, hence proposing here that there be a common EdgeX way defined to do this.
+
+### Target Users
+- Product Developers
+- Device Owner
+- Device User
+- Device Maintainer
+- Cloud User
+- Service Provider
+- Software Integrator
+
+### Description
+Picture the extremely simple case of a south-bound sensor device that just measures Temperature and Humidity and provides these as Device Resources. If we then add analytics and north-bound microservices:
+- A Trending service that needs Device Resources to indicate that Temperature and Humidity are trended for, eg, Minimum, Average, and Maximum over a 1 hour trend interval.
+- An Alarming service that needs Device Resources to describe the Alarm Rules used to monitor Temperature and Humidity, plus a device-level InAlarm status.
+- A Cloud service that reports not just the Temperature and Humidity but also their Trend configuration and Alarm Rule Resources. In addition, the Cloud service adds its own Resources to direct the Cadence with which this Device's data is reported.
+
+Now scale this up to 100 such Temperature/Humidity sensors, and it grows difficult to match all of the free-standing 
+Resources to their original sensor data. And add the requirement that all these resources must be able to be seen 
+and managed locally via REST or Message Bus, and potentially from north-bound services like Modbus/TCP, and from 
+the Cloud (because everybody wants to control everything from the Cloud). 
+
+Furthermore, from the end user's point of view, the Trend configuration, Alarm Rules, and Cloud Cadence that are added for a given Device are all seen as aspects of the Temperature/Humidity Device, as is common 
+for Digital Twin representations, and not as separated, free-standing entities. 
+So there must be some means to relate the extended Device Resources to the original south-bound Device and its
+Device Resources.
+
+
+### Existing solutions
+In EdgeX today, Devices and their Resources such as those described in the last section can be added, but they are not
+seen as related to the south-bound Device or to each other, except perhaps by well-chosen Labels or Tags.
+
+The existing south-bound Device Profiles could be extended to simply add the new Resources, but nothing connects these
+Resources to their owning Service (ie, so core-command could be used to manage them).
+
+
+### Requirements
+1. A means is defined to extend the Device metadata of a south-bound Device's profile with new resources that are 
+added and managed by an upper-level service, such as an analytics, utility, or north-bound service.
+2. The services which extend the device resources must manage the data for those extended resources on a per device instance basis.
```

Comment:
do we want to limit which services can add / extend?  Otherwise (impl detail), we need to have an API interface that all services abide in order for command to direct requests to?

Reply:
I think that the regular GET/PUT /device/name/{name}/{command} APIs for device resources are required. The third requirement about being able to set via core-command makes that so.